frontend/src/modules/EditZing/Components/StudentCard.tsx

```diff
@@ -33,6 +39,38 @@ const StudentCard = ({
     setSelected(!selected)
   }
 
+  const studentNotes = student.groups.find((g) => g.courseId === courseId)!
+    .notes
+  const [modalNotes, setModalNotes] = useState(studentNotes)
+
+  const [openNotesModal, setOpenNotesModal] = useState(false)
+  const [isNotesSaving, setIsNotesSaving] = useState(false)
+  const [showNotesSaveSuccess, setShowNotesSaveSuccess] = useState(false)
+  const [showNotesSaveFailure, setShowNotesSaveFailure] = useState(false)
+  const handleOpenNotes = () => {
+    setModalNotes(studentNotes) // Reset student notes every time modal opened
+    setOpenNotesModal(true)
+  }
+  const handleCloseNotes = () => setOpenNotesModal(false)
+
+  // saving notes to db
+  const saveModalNotes = async () => {
+    setIsNotesSaving(true)
+    await axios
+      .post(`${API_ROOT}${STUDENT_API}/notes`, {
+        email: student.email,
+        courseId: courseId,
+        notes: modalNotes,
+      })
+      .then(() => {
+        updateNotes(student.email, modalNotes) // Share the changes to rest of the page
+        setShowNotesSaveSuccess(true)
+      })
+      .catch(() => setShowNotesSaveFailure(true))
+    setIsNotesSaving(false)
+    setOpenNotesModal(false)
```

Comment:
Is this await not redundant? If you want to do it the await way, I think you can wrap the whole thing in a try catch and move all the elements around accordingly afterwards (i.e. catch into the catch block, then in the try.

Reply:
I believe Changyuan did this function ? Unless I did it and forgot

Reply:
@Coreball ? lol

Reply:
The only change I made to this section of code was adding `updateNotes(student.email, modalNotes)`. Otherwise it is essentially unchanged from the previous iteration, save for some variable/function renaming: https://github.com/cornell-dti/zing-lsc/blob/e126870bb3f0340690862ec284d37d998bc9c4bc/frontend/src/modules/EditZing/Components/NotesModal.tsx#L25-L40

Reply:
ye i mean i'll just remove the async from it all together

Reply:
Removing the async/await here broke the loading button on the modal because now it doesn't wait for the API call to finish before changing `isNotesSaving` back to false. The modal disappears instantly after pressing the save button.